sys/compat/freebsd64/freebsd64_uipc.c

```diff
@@ -254,31 +254,120 @@ freebsd64_copy_msg_out(struct msghdr *msg, struct mbuf *control)
 	return (error);
 }
 
+static int
+freebsd64_copyin_control(struct mbuf **mp, char * __capability buf,
+    u_int buflen)
+{
+	int error;
+	struct cmsghdr *cmsg;
+	struct mbuf *m = NULL;
+	caddr_t md;
+	u_int idx, newlen, msglen, datalen;
+
+	buflen = FREEBSD64_ALIGN(buflen);
+	if (buflen > MCLBYTES)
+		return (EINVAL);
+
+	/*
+	 * Iterate over the message headers to compute the new length using
+	 * the native kernel padding.
+	 */
+	idx = 0;
+	newlen = 0;
+	while (idx < buflen) {
+		cmsg = (struct cmsghdr *)((__cheri_fromcap char *)buf + idx);
+		msglen = fuword32(__USER_CAP_OBJ(&cmsg->cmsg_len));
+		if (msglen < sizeof(struct cmsghdr) ||
+		    idx + FREEBSD64_ALIGN(msglen) > buflen)
+			return (EINVAL);
+		datalen = (char *)cmsg + msglen - FREEBSD64_CMSG_DATA(cmsg);
+		idx += FREEBSD64_CMSG_SPACE(datalen);
+		newlen += CMSG_SPACE(datalen);
+	}
+
+	if (newlen > MCLBYTES)
```

Comment:
We might want to log this event if it happens as it's a potentially confusing deviation from native 64-bit.

Reply:
We should probably log the freebsd32 variant as well. What is the preferred way to log these things?